From: Martin K. Petersen <mkp@mkp.net>
Date: Thu, 8 Mar 2012 10:48:29 -0500
Subject: [PATCH] [SCSI] scsi_debug: Fix incorrect page length in logical block provisioning VPD
Git-commit: 3f0bc3b331a371392bb64c5b211b60ec84d5a444
References: FATE#313629
Patch-Mainline: v3.4

The page length for the 0xb2 VPD page is defined to be 4 bytes when no
provisioning descriptors are provided (DP=0).

Signed-off-by: Martin K. Petersen <martin.petersen@oracle.com>
Acked-by: Douglas Gilbert <dgilbert@interlog.com>
Signed-off-by: James Bottomley <JBottomley@Parallels.com>
Acked-by: Hannes Reinecke <hare@suse.de>
---
 drivers/scsi/scsi_debug.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/scsi/scsi_debug.c b/drivers/scsi/scsi_debug.c
index e4ba924..255570d 100644
--- a/drivers/scsi/scsi_debug.c
+++ b/drivers/scsi/scsi_debug.c
@@ -780,7 +780,7 @@ static int inquiry_evpd_b1(unsigned char *arr)
 /* Logical block provisioning VPD page (SBC-3) */
 static int inquiry_evpd_b2(unsigned char *arr)
 {
-	memset(arr, 0, 0x8);
+	memset(arr, 0, 0x4);
 	arr[0] = 0;			/* threshold exponent */
 
 	if (scsi_debug_lbpu)
@@ -795,7 +795,7 @@ static int inquiry_evpd_b2(unsigned char *arr)
 	if (scsi_debug_lbprz)
 		arr[1] |= 1 << 2;
 
-	return 0x8;
+	return 0x4;
 }
 
 #define SDEBUG_LONG_INQ_SZ 96
-- 
1.7.4.2

